### PR TITLE
Fix Issue1056

### PR DESCRIPTION
**Pull request by Anjali-NEC**
Fix issue #1056

### CODE REVIEW

File: src/lib/common/sem.cpp [hunk 1]
```diff
@@ -551,21 +550,14 @@ const char* timeStatSemGet(void)
 
 
 /* ****************************************************************************
```
Comment (fgalan) [inline src/lib/common/sem.cpp:552]: You have leaved the following in sem.cpp, but I think should be also moved:

* contexts_mutex_taken
* endpoint_mutexes_taken
* connectionContextSemGet()
* connectionSubContextSemGet()

and this file ends in line 549

Or maybe I'm missing something and they should be keep here?

Comment (fgalan) [reply src/lib/common/sem.cpp:552]: In fact in seems that contexts_mutex_taken and endpoint_mutexes_taken are already in the new file, so it is just a matter or removing it from src/lib/common/sem.cpp

File: src/lib/rest/curl.cpp [hunk 2]
```diff
@@ -0,0 +1,335 @@
+/*
```
Comment (fgalan) [inline src/lib/rest/curl.cpp:1]: Maybe better name names:

src/lib/rest/curlSem.cpp

as plain "curl" is too broad.

File: src/lib/rest/curl.cpp [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+#include "logMsg/traceLevels.h"
+
+#include "common/clockFunctions.h"
+#include "curl.h"
+
+
+
+/* ****************************************************************************
+*  curl context
+*/
```
Comment (fgalan) [inline src/lib/rest/curl.cpp:40]:
```suggestion
```

As there is no need of distinguish the set of functions that follow this mark within a bigger file, as in the original place.

File: src/lib/rest/curl.cpp (hunk 2 above)
Comment (fgalan) [reply src/lib/rest/curl.cpp:1]: ...and similar comment for curl.h -> curlSem.h

File: CHANGES_NEXT_RELEASE [hunk 4]
```diff
@@ -1,3 +1,4 @@
+- Fix: Cleanup lib/common/sem.cpp (#1056)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE:1]:
```suggestion
```

Except in special cases, refactors like this one are not mentioned in the CHANGES_NEXT_RELEASE file, as they are internal and for Orion user (the one to which changelogs are addressed) is transparent.

Comment (Anjali-NEC) [reply CHANGES_NEXT_RELEASE]: Fixed in https://github.com/telefonicaid/fiware-orion/pull/3869/commits/e4482579bd6b8125c68f3260d44d94f922e2cac6

File: src/lib/rest/curl.cpp (hunk 3 above)
Comment (Anjali-NEC) [reply src/lib/rest/curl.cpp]: Fixed in https://github.com/telefonicaid/fiware-orion/pull/3869/commits/5f663bcfa195d8da60f4c74e939e7d8c19b4bb9d

File: src/lib/rest/curl.cpp (hunk 2 above)
Comment (Anjali-NEC) [reply src/lib/rest/curl.cpp]: Fixed in https://github.com/telefonicaid/fiware-orion/pull/3869/commits/5f663bcfa195d8da60f4c74e939e7d8c19b4bb9d

File: src/lib/common/sem.cpp (hunk 1 above)
Comment (Anjali-NEC) [reply src/lib/common/sem.cpp]: Fixed in https://github.com/telefonicaid/fiware-orion/pull/3869/commits/5f663bcfa195d8da60f4c74e939e7d8c19b4bb9d

**Review by fgalan — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**